addons/mrp/static/src/widgets/timer.js

```diff
@@ -0,0 +1,66 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+const { Component, useState, onWillUpdateProps, onWillStart, onWillPatch, onWillDestroy } = owl;
+
+export class MrpTimer extends Component {
+    setup() {
+        super.setup();
+        this.state = useState({
+            hours: this.props.hours !== undefined ? this.props.hours : 0,
+            minutes: this.props.minutes !== undefined ? this.props.minutes : 0,
+            seconds: this.props.seconds !== undefined ? this.props.seconds : 0,
+        });
+
+        this.ongoing = this.props.ongoing;
+        if (this.props.record.data) {
+            this.ongoing = this.props.record.data.is_user_working;
+            this.state.seconds = this.props.record.data.duration * 60;
+        }
+        this._updateTimer();
+        onWillStart(() => this._runTimer());
+        onWillUpdateProps((nextProps) => {
+            this.ongoing = nextProps.ongoing;
+            this._runTimer();
+        });
+        onWillPatch(() => this._updateTimer());
+        onWillDestroy(() => clearTimeout(this.timer));
+    }
+
+    get toString() {
+        return _.str.sprintf(
```

Comment:
You can import and use the `sprintf` from the utils:
```javascript
import { sprintf } from '@web/core/utils/strings';
``` 
Like for jQuery, it's better to avoid using Underscore.js when we can